app/lib/frontend/models.dart

```diff
@@ -239,6 +239,54 @@ abstract class SecretKey {
   ];
 }
 
+/// An active invitation sent to a recipient.
+/// The parent entity is a [Package] the id is the concatenation of '[type]/[recipientEmail]'.
+///
+/// The invitation secret ([urlNonce]) is sent via e-mail to the recipient and
+/// they need to open a URL to accept the invitation.
+@db.Kind(name: 'PackageInvite', idType: db.IdType.String)
+class PackageInvite extends db.Model {
+  @db.StringProperty()
+  String type;
+
+  @db.StringProperty()
+  String recipientEmail;
+
+  @db.StringProperty()
+  String urlNonce;
+
+  @db.StringProperty()
+  String fromEmail;
+
+  @db.DateTimeProperty()
+  DateTime created;
+
+  @db.DateTimeProperty()
+  DateTime expires;
+
+  @db.DateTimeProperty()
+  DateTime lastNotified;
+
+  @db.IntProperty()
+  int notificationCount;
+
+  @db.DateTimeProperty()
+  DateTime confirmed;
+
+  /// Create a composite id.
+  static String createId(String type, String recipientEmail) =>
+      '$type/$recipientEmail';
+
+  bool isExpired() => new DateTime.now().toUtc().isAfter(expires);
+
+  /// Whether a new notification should be sent.
+  bool shouldNotify() {
+    final delay = Duration(minutes: 1 << notificationCount);
```

Comment:
Can this overflow and wrap around to become small again?

Reply:
Ah. So we only create a new notification *after* the timeout is over. :)
Otherwise it would be easy.

Reply:
I'm going to change this and the InviteStatus to `DateTime nextNotification`, as it would clear the ambiguity of the piece of code.